UI/LinkText.cpp

```diff
@@ -93,6 +94,51 @@ namespace {
 
         return retval;
     }
+
+    const std::string FOCS_VALUE_TAG_OPEN_PRE("<" + VarText::FOCS_VALUE_TAG);
+    const std::string FOCS_VALUE_TAG_CLOSE("</" + VarText::FOCS_VALUE_TAG + ">");
+    const xpr::sregex FOCS_VALUE_SEARCH = FOCS_VALUE_TAG_OPEN_PRE >> xpr::_s >> (xpr::s1 = REGEX_NON_BRACKET) >> ">" >>
+                                          (xpr::s2 = REGEX_NON_BRACKET) >> FOCS_VALUE_TAG_CLOSE;
+
+    /** Parses VarText::FOCS_VALUE_TAG%s within @p text, replacing value ref name%s with
+     *  the evaluation result of that value ref.  If link label is empty, inserts resolved link argument as label */
+    std::string ValueRefLinkText(const std::string& text, const bool useDescriptionInsteadOfUserString) {
+        if (!boost::contains(text, FOCS_VALUE_TAG_CLOSE))
+            return text;
+
+        std::string retval(text);
+        auto text_it = retval.begin();
+        xpr::smatch match;
+
+        while (true) {
+            if (!xpr::regex_search(text_it, retval.end(), match, FOCS_VALUE_SEARCH, xpr::regex_constants::match_default))
+                break;
+
+            std::string value_ref_name(match[1]);
+            auto        value_ref = GetAnyValueRef(value_ref_name);
+            std::string value_str = value_ref_name;
+            std::string explanation_str(match[2]);
```

Comment:
ideally wrap with `(` and `)` here rather than later, to avoid extra string allocations and to simplify the later code at lines 122-131 where you'd be able to remove several `else` cases that just do the wrapping there.